PixelHunter1995/Components/SpeechLine.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+
+namespace PixelHunter1995.Components
+{
+    internal class SpeechLine : IUpdateable
+    {
+        private string ActiveSpeech;
+        private TimeSpan SpeechEndTime;
+        public bool Speaking { get; private set; }
+        private bool ShouldStartNextLine;
+        private readonly Queue UpcomingLines;
+        private static readonly int TALKING_SPEED = 100;  // Number of milliseconds per character in line
+        private static readonly int CHARS_PER_LINE = 40;
+
+        public SpeechLine()
+        {
+            ActiveSpeech = "";
+            Speaking = false;
+            ShouldStartNextLine = false;
+            UpcomingLines = new Queue();
+        }
+
+        public void Say(string speech)
+        {
+            ShouldStartNextLine = true;
+            UpcomingLines.Enqueue(speech);
+        }
+
+        public void Update(GameTime gameTime)
+        {
+            if (!Speaking || ShouldStartNextLine)
+            {
+                if (UpcomingLines.Count != 0)
+                {
+                    NextLine(gameTime);
+                }
+                else
+                {
+                    Speaking = false;
+                }
+            }
+            else if (SpeechEndTime < gameTime.TotalGameTime)  // Done with this line
+            {
+                ShouldStartNextLine = true;
+            }
+        }
+
+        private void NextLine(GameTime gameTime)
+        {
+            ActiveSpeech = (string) UpcomingLines.Dequeue();
+            var speechDuration = new TimeSpan(0, 0, 0, 0, TALKING_SPEED * ActiveSpeech.Length);
+            SpeechEndTime = gameTime.TotalGameTime + speechDuration;
+            ShouldStartNextLine = false;
+            Speaking = true;
```

Comment:
Maybe I misunderstand the control flow here, but it feels like setting `Speaking` to true here should be unnecessary? I would think that `Speaking` would be set to true when `Say()` is invoked (which it is not... I guess I'm misunderstanding something). But then, maybe `Speaking` should only be a function of if there are lines left in `UpcomingLines`?